registry/storage/catalog.go

```diff
@@ -93,5 +99,38 @@ func (reg *registry) Enumerate(ctx context.Context, ingester func(string) error)
 		}
 	}
 	return nil
+}
+
+// todo(richardscothern): find a better place for this
+func (reg *metadataRegistry) Repositories(ctx context.Context, repos []string, lastRepo string) (n int, errVal error) {
+	if len(repos) == 0 {
+		return 0, errors.New("no space in slice")
+	}
+
+	var items []string
+
+	params := metadata.IterateParameters{From: lastRepo, IterType: metadata.RepoKey{}}
+	err := reg.metadataService.Iterate(ctx, params, func(key string, val interface{}) error {
+		items = append(items, key)
+		if len(items) >= len(repos) {
+			// filled the give slice
+			return ErrFinishedWalk
+		}
+		return nil
+	})
+
+	if err != nil && err != ErrFinishedWalk {
+		return 0, err
+	}
+
+	n = copy(repos, items)
+	if len(items) < len(repos) {
+		// didn't fill the repos array, no more to return
```

Comment:
Is there no other indicator of `io.EOF` or by returning at `len(items) == len(repos)` any indicator of completeness by the `Iterate` will not be reached.